osquery/core/process.h

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <string>
+#include <boost/optional.hpp>
+
+#include <osquery/core.h>
+
+namespace osquery {
+
+#ifdef WIN32
+using PlatformPidType = HANDLE;
+#else
+using PlatformPidType = pid_t;
+#endif
+
+const PlatformPidType kInvalidPid = (PlatformPidType) -1;
+
+enum ProcessState {
+  PROCESS_ERROR           = -1,
+  PROCESS_STILL_ALIVE     =  0,
+  PROCESS_EXITED,
+  PROCESS_STATE_CHANGE
+};
+
+class PlatformProcess {
+  public:
+    PlatformProcess(): id_(kInvalidPid) { }
+    PlatformProcess(PlatformPidType id);
+    PlatformProcess(const PlatformProcess& src);
```

Comment:
Do we need these? Can we consider a `PlatformProcess` to be non-copyable? Then we force callsites to pass around shared pointers if they need to keep member instances.